### PR TITLE
Gh-3203: GafferPop filter edges by label

### LINKED ISSUE #3203: GafferPop not filtering edges by label
**Describe the bug**
The following gremlin query fails in the Cucumber test suite
```
g.V().choose(__.out('knows').count().is(P.gt(0)),
             __.out("knows"),
             __.identity())
      .values('name')
```
This is due to out('knows') not filtering the edges

### PR DESCRIPTION
**Pull request by p29876**
# Related issue

- Resolve #3203

### CODE REVIEW

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertex.java [hunk 1]
```diff
@@ -163,16 +163,12 @@ public Edge addEdge(final String label, final Vertex vertex, final Object... key
...
-                return StreamSupport.stream(edgeVertexes.spliterator(), false)
-                    .anyMatch(v -> ElementHelper.areEqual(v, this));
-            })
-            .iterator();
+        // Get edges with matching labels and direction from the graph
+        View view = new View.Builder()
+            .edges(Arrays.asList(edgeLabels))
+            .build();
+
+        return this.edges(direction, view);
```
Comment (tb06904) [inline library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertex.java:171]: Super minor but probably don't need the `this`

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java [hunk 2]
```diff
@@ -241,7 +242,10 @@ void shouldDelegateEdgesToGraph() {
         final GafferPopGraph graph = mock(GafferPopGraph.class);
         final GafferPopVertex vertex = new GafferPopVertex(TestGroups.ENTITY, GafferPopGraph.ID_LABEL, graph);
         final Iterable<Edge> resultEdges = Arrays.asList(((Edge) new GafferPopEdge(GafferPopGraph.ID_LABEL, vertex, vertex, graph)));
-        given(graph.edges(GafferPopGraph.ID_LABEL, new String[]{TestGroups.ENTITY})).willReturn(resultEdges.iterator());
+        given(graph.edgesWithView(GafferPopGraph.ID_LABEL, Direction.IN, new View.Builder()
+                .edges(List.of(TestGroups.ENTITY))
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:246]: This doesn't compile. [`Collections.singletonList`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singletonList-T-) should work instead. Or just `Arrays.asList` which is already imported.
```suggestion
                .edges(Collections.singletonList(TestGroups.ENTITY))
```

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java [hunk 3]
```diff
@@ -241,11 +241,11 @@ void shouldGetOutgoingEdges() {
         // Then
         assertThat(result)
                 .extracting(item -> item.id().toString())
-                .containsExactly("[p1, p2]", "[p1, s1]");
+                .containsExactlyInAnyOrder("[p1, p2]", "[p1, s1]");
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java:244]: Why is this change required? Does this mean that it was in a defined order before but now no longer is?

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java (hunk 2 above)
Comment (p29876) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:246]: Yep, just spotted this! I had Java 11 installed for some reason

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java (hunk 3 above)
Comment (p29876) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java:244]: @GCHQDeveloper314 changing the filtering appears to have just swapped the order of the returned edges. If the order doesn't matter then tempted to leave as is

Comment (GCHQDeveloper314) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java:244]: Ok, that's probably fine then.

**Review by cn337131 — APPROVED** [review]

**Review by tb06904 — APPROVED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**